crates/compiler/src/compilation_steps/generate_code.rs

```diff
@@ -0,0 +1,102 @@
+use crate::listeners::CompilerListener;
+use crate::prelude::generated::yarnspinnerparser::YarnSpinnerParserTreeWalker;
+use crate::prelude::*;
+use crate::visitors::KnownTypes;
+use std::collections::{HashMap, HashSet};
+
+pub(crate) fn generate_code(mut state: CompilationIntermediate) -> CompilationIntermediate {
+    let has_errors = state
+        .diagnostics
+        .iter()
+        .any(|d| d.severity == DiagnosticSeverity::Error);
+    let results: Vec<_> = if has_errors {
+        // We have errors, so we can't safely generate code.
```

Comment:
fine with this for now, but I wonder if we should make the previous result return a Result instead of having to explicitly check for errors?